polyfill/lib/date.mjs

```diff
@@ -256,18 +256,27 @@ export class Date {
     const year = GetSlot(this, ISO_YEAR);
     const month = GetSlot(this, ISO_MONTH);
     const day = GetSlot(this, ISO_DAY);
-    const calendar = GetSlot(this, CALENDAR);
+    const dateCalendar = GetSlot(this, CALENDAR);
     const DateTime = GetIntrinsic('%Temporal.DateTime%');
 
-    if (temporalTime === undefined) return new DateTime(year, month, day, 0, 0, 0, 0, 0, 0, calendar);
+    if (temporalTime === undefined) return new DateTime(year, month, day, 0, 0, 0, 0, 0, 0, dateCalendar);
 
     temporalTime = ES.ToTemporalTime(temporalTime, GetIntrinsic('%Temporal.Time%'));
-    const hour = GetSlot(temporalTime, HOUR);
-    const minute = GetSlot(temporalTime, MINUTE);
-    const second = GetSlot(temporalTime, SECOND);
-    const millisecond = GetSlot(temporalTime, MILLISECOND);
-    const microsecond = GetSlot(temporalTime, MICROSECOND);
-    const nanosecond = GetSlot(temporalTime, NANOSECOND);
+    const hour = GetSlot(temporalTime, ISO_HOUR);
+    const minute = GetSlot(temporalTime, ISO_MINUTE);
+    const second = GetSlot(temporalTime, ISO_SECOND);
+    const millisecond = GetSlot(temporalTime, ISO_MILLISECOND);
+    const microsecond = GetSlot(temporalTime, ISO_MICROSECOND);
+    const nanosecond = GetSlot(temporalTime, ISO_NANOSECOND);
+    const timeCalendar = GetSlot(temporalTime, CALENDAR);
+
+    let calendar;
+    if (dateCalendar.toString() !== timeCalendar.toString() && dateCalendar.toString() === 'iso8601') {
```

Comment:
Use `ES.CalendarEquals()` and `ES.CalendarToString()` here.

I'm also not sure the logic is correct - what if the calendars are different and neither of them are ISO? I think it should throw in that case.